miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/rest/BoxRestController.java

```diff
@@ -609,13 +615,39 @@ private static boolean isRealBarcode(BoxScan scan, String barcode) {
     return !barcode.equals(scan.getNoTubeLabel()) && !barcode.equals(scan.getNoReadLabel());
   }
 
-  @RequestMapping(value = "/box/{boxId}/discard-all")
-  @ResponseStatus(HttpStatus.NO_CONTENT)
-  public void discardEntireBox(@PathVariable long boxId) throws IOException {
+  @DeleteMapping("/box/{boxId}/positions/{position}")
+  public @ResponseBody BoxDto removeSingleItem(@PathVariable long boxId, @PathVariable String position) throws IOException {
+    Box box = getBox(boxId);
+    box.getBoxPositions().remove(position);
+    boxService.save(box);
+    return getBoxDtoWithBoxables(boxId);
+  }
+
+  @PostMapping("/box/{boxId}/positions/{position}/discard")
+  public @ResponseBody BoxDto discardSingleItem(@PathVariable long boxId, @PathVariable String position) throws IOException {
+    Box box = getBox(boxId);
+    boxService.discardSingleItem(box, position);
+    return getBoxDtoWithBoxables(boxId);
+  }
+
+  private Box getBox(long boxId) throws IOException {
     Box box = boxService.get(boxId);
     if (box == null) {
       throw new RestException("Box " + boxId + " not found", Status.NOT_FOUND);
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal " not found" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)